webapp/tests/forms/steps/lotse/test_steuerminderungen.py

```diff
@@ -16,6 +23,184 @@ def step_gem_haushalt():
+    def test_if_religion_not_set_then_raise_validation_error(self):
+        data = {}
+        with pytest.raises(ValidationError):
+            ShowReligionPrecondition.parse_obj(data)
+
+    def test_if_religion_set_false_then_raise_validation_error(self):
+        data = {'stmind_select_religion': False}
+        with pytest.raises(ValidationError):
+            ShowReligionPrecondition.parse_obj(data)
+
+    def test_if_religion_set_true_then_do_not_raise_validation_error(self):
+        data = {'stmind_select_religion': True}
+        try:
+            ShowReligionPrecondition.parse_obj(data)
+        except ValidationError:
+            pytest.fail("Should not raise a validation error")
+
+    def test_step_to_redirect_to_is_set_correctly(self):
+        assert ShowReligionPrecondition._step_to_redirect_to == StepSelectStmind.name
+
+
+class TestNotShowPersonBPrecondition:
+    def test_if_show_person_b_true_then_raise_validation_error(self):
+        with patch('app.model.form_data.JointTaxesModel.show_person_b', return_value=True), \
+                pytest.raises(ValidationError):
+            NotShowPersonBPrecondition.parse_obj({'familienstand': 'single'})
+
+    def test_if_show_person_b_false_then_do_not_raise_validation_error(self):
+        try:
+            with patch('app.model.form_data.JointTaxesModel.show_person_b', return_value=False):
+                NotShowPersonBPrecondition.parse_obj({'familienstand': 'single'})
+        except ValidationError:
+            pytest.fail("Should not raise a validation error")
+
+    def test_step_to_redirect_to_is_set_correctly(self):
+        assert NotShowPersonBPrecondition._step_to_redirect_to == StepFamilienstand.name
+
+
+class TestHandwerkerHaushaltsnaheSetPrecondition:
+    def test_if_haushaltsnahe_and_handwerker_not_set_then_raise_validation_error(self):
+        data = {}
+        with pytest.raises(ValidationError):
+            HandwerkerHaushaltsnaheSetPrecondition.parse_obj(data)
+
+    def test_if_haushaltsnahe_summe_set_then_do_not_raise_validation_error(self):
+        data = {'stmind_haushaltsnahe_summe': 30.0}
+        try:
+            HandwerkerHaushaltsnaheSetPrecondition.parse_obj(data)
+        except ValidationError:
+            pytest.fail("Should not raise a validation error")
+
+    def test_if_handwerker_summe_set_then_do_not_raise_validation_error(self):
+        data = {'stmind_handwerker_summe': 30.0}
+        try:
+            HandwerkerHaushaltsnaheSetPrecondition.parse_obj(data)
+        except ValidationError:
+            pytest.fail("Should not raise a validation error")
+
+    def test_step_to_redirect_to_is_set_correctly(self):
+        assert HandwerkerHaushaltsnaheSetPrecondition._step_to_redirect_to == StepHaushaltsnaheHandwerker.name
+
+
+# STEPS
+
+@pytest.mark.usefixtures('test_request_context')
+class TestStepSelectStmind:
+    def test_no_precondition_set(self):
+        assert StepSelectStmind.preconditions == []
```

Comment:
The idea about not testing data applies here, too, same as for testing `_step_to_redirect_to`…

Also, why does this require a `test_request_context`? Is it doing anything that I'm missing just by looking at this line? Maybe I'm just confused! :D

Reply:
No, the `test_request_context` ist not necessary. 

The same reason of adding the tests applies here too: It just feels weird to a) test the precondition itself and then b) never again that it is used anywhere - except for the functional tests.